src/player_event_handler.rs

```diff
@@ -0,0 +1,50 @@
+use std::process::Command;
+use std::sync::mpsc::{channel, Sender};
+use std::thread;
+use librespot::playback::config::PlayerEvent;
+
+fn run_program(program: &str, args: Vec<String>) {
+    info!("Running {}", program);
+    let mut v: Vec<&str> = program.split_whitespace().collect();
+    let status = Command::new(&v.remove(0))
+        .args(&v)
+        .args(args)
+        .status()
+        .expect("program failed to start");
+    info!("Exit status: {}", status);
+}
+
+pub fn run_program_on_events(onstart: Option<String>, 
+                             onstop: Option<String>,
+                             onchange: Option<String>) -> Option<Sender<PlayerEvent>> {
+    if onstart.is_none() && onstop.is_none() && onchange.is_none() {
+        None
+    } else {
+        let (sender, receiver) = channel();
+        thread::spawn(move || {
+            while let Ok(msg) = receiver.recv() {
+                match msg {
+                    PlayerEvent::Changed { old_track_id, new_track_id } => {
+                        let args = vec![old_track_id.to_base16(), new_track_id.to_base16()];
```

Comment:
Encoding in base62 is better, as it's the usual "spotify url" format.

Reply:
I agree, but perhaps this can wait until `u128` lands on stable? That would make the implementation a lot simpler.